### PR TITLE
CMake NetCDF fixes

### PR DESCRIPTION
**Pull request by naromero77**
NetCDF include directory needs to be passed into CMake. Add support for Tulip.

### CODE REVIEW

**Comment by naromero77** [comment]
Changes have also been tested on Summit with CUDA 11.0.3 which includes CUB as part of the distribution.

**Comment by naromero77** [comment]
ping

File: cpp/test/summit_gpu.sh [hunk 1]
```diff
@@ -4,10 +4,12 @@ source $MODULESHOME/init/bash
...
 module load DefApps gcc cmake cuda netcdf-cxx4 netcdf nco
 
-export NCFLAGS="`ncxx4-config --libs`"
+export NCINCLUDE="`ncxx4-config --includedir`;`nc-config --includedir`" # must be semi-colon seperated
+export NCFLAGS="`ncxx4-config --libs` `nc-config --libs`"
 export CC=gcc
 export CXX=g++
 export CXXFLAGS="-O3"
 export ARCH="CUDA"
 export CUDA_ARCH="-arch sm_70 --std=c++14 --use_fast_math -O3"
```
Comment (brhillman) [inline cpp/test/summit_gpu.sh:13]: Why did you add the CUBHOME variable? This does not appear to be used.

Comment (naromero77) [reply cpp/test/summit_gpu.sh:13]: YAKL needs it to build properly for the CUDA backend.

Comment (brhillman) [reply cpp/test/summit_gpu.sh:13]: Are you sure? I'm not setting a `CUBHOME` variable on Summit, and I seem to be able to build and run fine with a CUDA backend. And I don't see this environment variable anywhere in the RRTMGP or YAKL source. @mrnorman can you comment on this?
